File: runtime_permission/src/main/java/net/crystalapps/permission/runtime/library/utils/ObjectUtil.java
package net.crystalapps.permission.runtime.library.utils;

/**
 * Created by Syed Owais Ali on 10/13/2018.
 */
@SuppressWarnings("WeakerAccess")
public class ObjectUtil {

    public static <T> T requireNonNull(T obj) {
        return requireNonNull(obj, null);
    }

    public static <T> T requireNonNull(T obj, String message) {
        if (obj == null)
            throw new NullPointerException(message);
        return obj;
    }
}

File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/utils/ObjectUtil.kt
package net.crystalapps.permission.runtime.library.utils

/**
 * Created by Syed Owais Ali on 10/13/2018.
 */
object ObjectUtil {

    fun <T> requireNonNull(obj: T): T {
        return requireNonNull(obj, null)
    }

    fun <T> requireNonNull(obj: T?, message: String?): T {
        if (obj == null)
            throw NullPointerException(message)
        return obj
    }
}
